### PR TITLE
[Backport-v2] Added imagePullSecrets field in otel events and Sumologic-setup template

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Backport of #2689 to v2

### CODE REVIEW

File: deploy/helm/sumologic/templates/setup/job.yaml [hunk 1]
```diff
@@ -52,6 +52,7 @@ spec:
       containers:
       - name: setup
         image: {{ .Values.sumologic.setup.job.image.repository }}:{{ .Values.sumologic.setup.job.image.tag }}
+        imagePullSecrets: {{ .Values.sumologic.setup.job.image.pullSecrets }}
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/templates/setup/job.yaml:55]: What about this? https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/v2.19.0/deploy/docs/Working_with_container_registries.md?plain=1#L46

**Review by astencel-sumo — CHANGES_REQUESTED** [review]
The setup job already has this

File: deploy/helm/sumologic/templates/events/otelcol/statefulset.yaml [hunk 2]
```diff
@@ -100,6 +100,10 @@ spec:
       containers:
       - name: otelcol
         image: {{ .Values.otelevents.image.repository }}:{{ .Values.otelevents.image.tag }}
+{{- if .Values.sumologic.pullSecrets }}
+        imagePullSecrets:
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/templates/events/otelcol/statefulset.yaml:104]: Doesn't `otelevents` already use the `sumologic.pullSecrets` by using the common service account?

**Comment by swiatekm-sumo** [comment]
Reverting the original change in https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2721 and closing this.

**Pull request closed without merging**